lib/src/utils/serialization_utils.dart

```diff
@@ -33,6 +45,89 @@ class SerializationUtils {
     writer.writeUint32((value & maxUint32).toInt());
   }
 
+  static void serializeOptionFlag(ByteDataWriter writer, bool included) {
```

Comment:
instead of `serialize...`, please rename to `writeSerialized...`. Do this for all the methods that are directly writing something to `writer` (as opposed to returning the serialized value as function return)

Reply:
P.s. The functions returning `Uint8List`/etc should keep the name starting with `serialize...`